src/Resources/AbstractResource.php

```diff
@@ -8,11 +8,34 @@ abstract class AbstractResource
 {
     public function toArray(): array
     {
-        return (array) $this;
+        $array = [];
+
+        $this->__objectToArray($this, $array);
+
+        return $array;
     }
 
     public function toJson(): string
     {
         return json_encode($this);
     }
+
+    private function __objectToArray($obj, &$arr)
+    {
+        if (! is_object($obj) && ! is_array($obj)) {
```

Comment:
Why would this class not be an object or the empty array you just created not be an array?